clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java

```diff
@@ -222,6 +222,7 @@ public enum ConfigSource {
         DYNAMIC_BROKER_LOGGER_CONFIG,   // dynamic broker logger config that is configured for a specific broker
         DYNAMIC_BROKER_CONFIG,          // dynamic broker config that is configured for a specific broker
         DYNAMIC_DEFAULT_BROKER_CONFIG,  // dynamic broker config that is configured as default for all brokers in the cluster
+        DYNAMIC_CLIENT_METRICS_CONFIG,  // dynamic client metrics subscription config that is configured for all clients
```

Comment:
It seems weird to put `CLIENT_METRICS` here. The rest of the entries all seem to be topic config related. Other entity types like user, client, etc are not listed here. Ditto for the change in `DescribeConfigsResponse` and `ConfigResource`.

Reply:
I might be wrong but shouldn't this be needed to make `kafka-config.sh` work to create `subscription` for client_metrics. To support the `kafka-config.sh` utility we would require change in `ConfigEntry` and `ConfigSource`. I have this PR that uses same: https://github.com/apache/kafka/pull/14632

Reply:
@apoorvmittal10 : Thanks for the comment. I mis-understood the logic here. The create/describe config RPC only applies to topic/broker/BrokerLoggerConfigType and we are adding CLIENT_METRICS here. quota/Ip level configs are handled by different types of PRC. So, the logic you have here is fine.